bands.py

```diff
@@ -0,0 +1,69 @@
+import datetime
+from typing import Iterable
+import uuid
+
+from django.db import models
+from django.utils import timezone
+
+# region Models
+
+
+class BaseModel(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+
+class Band(BaseModel):
+    name = models.CharField(max_length=128)
+
+
+class Album(BaseModel):
+    band = models.ForeignKey(to="Band", on_delete=models.CASCADE)
+    name = models.CharField()
+    release = models.DateField()
+
+
+class Song(BaseModel):
+    album = models.ForeignKey(to="Album", on_delete=models.CASCADE)
+    name = models.CharField(max_length=128)
+    duration_seconds = models.PositiveIntegerField()
+
+
+class Artist(BaseModel):
+    name = models.CharField(max_length=128)
+
+
+class BandArtist(BaseModel):
+    band = models.ForeignKey(to="Band", on_delete=models.CASCADE)
+    artist = models.ForeignKey(to="Artist", on_delete=models.CASCADE)
+    joined = models.DateField()
+    left = models.DateField(null=True, blank=True)
+
+
+class Stream(BaseModel):
+    user = models.ForeignKey(to="auth.User", on_delete=models.SET_NULL)
+    song = models.ForeignKey(to="Song", on_delete=models.CASCADE)
+    stream_date = models.DateField()
+    duration_seconds = models.PositiveIntegerField()
+
+
+# endregion
+
+# region Utils
+
+
+def get_band_formation(band_id: str, date: datetime.date) -> Iterable[str]:
+    
+    band_artists: Iterable[BandArtist] = BandArtist.objects.filter(band_id=band_id, joined__lte=date, left__gt=date)
+    for band_artist in band_artists:
+        yield band_artist.artist
```

Comment:
Hi,

I found that you tried to get "artist" data from "band_artists". :)
This looks good.

I will suggest you to use another method that can only return the field that you want and save the memory.

For example:

```py
BandArtist.objects.filter(...).only('artist')
```